src/grpc.cc

```diff
@@ -170,9 +170,87 @@ static grpc::Status marshal_value_list(const value_list_t *vl,
   msg->set_allocated_time(new google::protobuf::Timestamp(t));
   msg->set_allocated_interval(new google::protobuf::Duration(d));
 
+  msg->clear_meta_data();
+  if (vl->meta != nullptr) {
```

Comment:
This `if` block is a bit too long. Could you break this out into a `marshal_meta_data()` function? Dito for unmarshal below.